web3/contract/base_contract.py

```diff
@@ -436,6 +437,7 @@ def __init__(
         w3: "Web3",
         contract_function_class: Type["BaseContractFunction"],
         address: Optional[ChecksumAddress] = None,
+        decode_tuples: Optional[bool] = None,
```

Comment:
Is there a variant where we're explicitly using `None` somewhere? If not, I think I'd prefer to just have this default to `False`, but feel free to take or leave.

Reply:
+1 for this... the `ccip_read_enabled` flag was `None` for a reason since there are two competing flags and only if the contract flag was explicitly set to a value would it be considered, having the global flag override its value otherwise. If there isn't a more complex situation here, the `False` default would be ideal.

Reply:
set to default to `False` in all contract files and in docs.